### ISSUE TITLE
[Bug]: cancel query takes very long

### ISSUE THREAD

**Issue by fengttt** [matrixorigin/matrixone#12324]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93): 1.0-dev mo cloud
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

A long running query, for example one takes 30 seconds, cannot be cancelled.   ^C won't return.  For example,
the following query takes more than 20 seconds and it is actually not cancelled.

mysql> select span_name, span_id, parent_span_id, duration from span_info order by duration desc;
^C^C -- query aborted
+----------------------+------------------+------------------+----------------+
| span_name            | span_id          | parent_span_id   | duration       |
+----------------------+------------------+------------------+----------------+
| MetricStorageUsage   | 6cb69fe6404f0829 | 1595185840e7252d | 58240822308681 |
+----------------------+------------------+------------------+----------------+
4146 rows in set (20.07 sec)

mysql>

**Comment by fengttt** [comment]
Sorry, git version - 
mysql> select git_version();
+---------------+
| git_version() |
+---------------+
| d65f651f7     |
+---------------+
1 row in set (0.53 sec)

**Comment by fengttt** [comment]
This has very big impact on user experience.

**Comment by YANGGMM** [comment]
plz @qingxinhome

**Comment by qingxinhome** [comment]
Processing

**Comment by qingxinhome** [comment]
经过断点日志分析，cancel长久不能生效的原因是projection算子和output算子，一直在执行且缺少cancel断言

**Comment by qingxinhome** [comment]
PR to be optimized

**Comment by qingxinhome** [comment]
association:https://github.com/matrixorigin/MO-Cloud/issues/481

**Comment by qingxinhome** [comment]
association: https://github.com/matrixorigin/MO-Cloud/issues/1579

**Comment by qingxinhome** [comment]
association: https://github.com/matrixorigin/MO-Cloud/issues/1579

**Comment by qingxinhome** [comment]
The issue has been resolved and PR has been submitted to main and 1.1 dev. Please verify and test @fengttt @sukki37 @aressu1985

**Comment by aressu1985** [comment]
fixed
